### DISCUSSION TITLE
analogReadEnh interrupts & overflow?

### DISCUSSION THREAD

**Discussion by grandaspanna** [SpenceKonde/megaTinyCore#675] [Q&A]
Loving the analogReadEnh in a new little project, where I'm using a precision voltage reference into a 1626 to provide accuracy down to a few mV.

Two questions:

1. Is it interruptable? Haven't yet timed it, but I'm driving a 4-digit seven segment display directly and at 16-bit precision, I'm getting a bit of flicker. I'm wondering if moving the display refresh to an interrupt would help.
2. Can I distinguish something over VREF, or does it just return the maximum possible value for the number of bits (less 1)?

**Comment by SpenceKonde** [comment]
It's like analogRead on the first point - it just sits there with a busy
wait loop until resrdy is set.

As to the second, nope, you can't tell if the voltage is higher than the
reference versus about the same as ref. Not without some other step, one
way or another

____________
Spence Konde
Azzy’S Electronics

New products! Check them out at tindie.com/stores/DrAzzy
GitHub: github.com/SpenceKonde
ATTinyCore: Arduino support for almost every ATTiny microcontroller
Contact: ***@***.***

**Reply by grandaspanna** [reply]
Thanks. I'll have a think about options.

**Reply by SpenceKonde** [reply]
You could pair it with the analogComparator to detect if the ADC input is higher that some value that's higher than the reference voltage, I would think....
